#! /bin/sh /usr/share/dpatch/dpatch-run
## 90_fix_forced_servicechecks.dpatch by Alexander Wirt <formorer@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes handling of forced checks

@DPATCH@
diff -urNad nagios3-3.0.6~/cgi/cmd.c nagios3-3.0.6/cgi/cmd.c
--- nagios3-3.0.6~/cgi/cmd.c	2008-11-30 18:22:58.000000000 +0100
+++ nagios3-3.0.6/cgi/cmd.c	2008-12-12 13:50:46.000000000 +0100
@@ -2033,7 +2033,8 @@
 		break;
 
 	case CMD_SCHEDULE_SVC_CHECK:
-	case CMD_SCHEDULE_FORCED_SVC_CHECK:
+		if (force_check == TRUE)
+			cmd = CMD_SCHEDULE_FORCED_SVC_CHECK;
 		result = cmd_submitf(cmd,"%s;%s;%lu",host_name,service_desc,start_time);
 		break;
 
